### ISSUE TITLE
line-by-line memory usage on a script not ending in .py

### ISSUE THREAD

**Issue by SyamGadde** [pythonprofilers/memory_profiler#268]
It seems as if the following usage:

```
$ python -m memory_profiler example.py
```

does not work if the script does not end with .py:

```
$ python -m memory_profiler /path/to/myscript
```

because memory_profiler looks for a module with the name /path/to/myscript.  Since the module memory_profiler is intending to execute needs an `if __name__ == '__main__'` clause anyway, is there any reason memory_profiler shouldn't just use the `exec_with_profiler` code path in all cases?  That would also make sure any typical python command line of the form `python scriptname` can take the `-m memory_profiler` option without needing to change syntax or file names.

Thanks for considering!

**Comment by fabianp** [comment]
I agree. Right now it's not clear to me why we are doing different things depending on whether the script ends in .py (see around ```if args.program[0].endswith('.py')``` . Pull requests welcomed
